src/anndata/_io/specs/methods.py

```diff
@@ -695,12 +695,11 @@ def write_recarray_zarr(
     from anndata.compat import _to_fixed_length_strings
 
     elem = _to_fixed_length_strings(elem)
```

Comment:
that function links to a PR that has been superseeded by https://github.com/zarr-developers/zarr-python/pull/813, which is merged.

while we’re cleaning up zarr recarray stuff: should this be handled differently?

Reply:
if recarrays don’t need fixed-size types for some reason, we should store strings inside of them the same way we store strings elsewhere, i.e. using the variable length codec.

Reply:
We can't get rid of this becasue `object` dtypes don't write through to `zarr` so different reason but same result, the need for conversion.  Maybe we could use numpy `StringDtype` instead though!